internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -1836,49 +1794,43 @@ func TestReconcileReferencedObjectSequences(t *testing.T) {
 						ControlPlane: &scope.ControlPlaneState{
 							Object: &unstructured.Unstructured{
 								Object: map[string]interface{}{
-									"kind":       "GenericControlPlane",
-									"apiVersion": "controlplane.cluster.x-k8s.io/v1beta1",
+									"kind":       builder.TestControlPlaneKind,
+									"apiVersion": builder.ControlPlaneGroupVersion.String(),
 									"metadata": map[string]interface{}{
 										"name":      "my-cluster",
 										"namespace": namespace.GetName(),
 									},
-									"spec": step.desired.spec,
+									// "spec": step.desired.spec,
```

Comment:
```suggestion
```
nit

Reply:
resolving as it is part of https://github.com/kubernetes-sigs/cluster-api/pull/6495#discussion_r891313460